lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -700,98 +739,72 @@ def _generate(self, context):
-                    ).format(
-                        scorable_block_name=scorable_block.display_name,
-                        assignment_type=assignment_type_name,
-                        subsection_index=subsection_index,
-                        subsection_name=subsection_info['subsection_block'].display_name,
-                    )
-                    scorable_blocks_map[scorable_block.location] = [header_name + " (Earned)",
-                                                                    header_name + " (Possible)"]
-        return scorable_blocks_map
-
-    def _rows_for_users(self, context, users, header_row, graded_scorable_blocks):
+    def _rows_for_users(self, context, users):
         """
         Returns a list of rows for the given users for this report.
         """
-        if six.text_type(context.course_id) == 'course-v1:MITx+CTL.SC0x+1T2019':
-            context.update_status('ProblemGradeReport Investigation log: Start of row creation for users')
+        self.log_additional_info_for_mit_course(context, 'ProblemGradeReport: Start of row creation for users')
         success_rows, error_rows = [], []
-        for student, course_grade, error in CourseGradeFactory().iter(users, context.course):
-            student_fields = [getattr(student, field_name) for field_name in header_row]
+        for student, course_grade, error in CourseGradeFactory().iter(
+            users,
+            course=context.course,
+            collected_block_structure=context.course_structure,
+            course_key=context.course_id,
+        ):
             context.task_progress.attempted += 1
-            if context.task_progress.attempted % 2 == 0 and \
-                    six.text_type(context.course_id) == 'course-v1:MITx+CTL.SC0x+1T2019':
-                context.update_status('ProblemGradeReport Investigation log: Processing user {}'.format(
-                    context.task_progress.attempted))
+            if context.task_progress.attempted % 2 == 0:
+                self.log_additional_info_for_mit_course(
+                    context,
+                    'ProblemGradeReport: Total Processed:{}'.format(context.task_progress.attempted)
+                )
             if not course_grade:
                 err_msg = text_type(error)
                 # There was an error grading this student.
                 if not err_msg:
                     err_msg = 'Unknown error'
-                error_rows.append(student_fields + [err_msg])
+                error_rows.append(
+                    [student.id, student.email, student.username] +
+                    [err_msg]
+                )
                 context.task_progress.failed += 1
                 continue
 
-            enrollment_status = _user_enrollment_status(student, context.course_id)
-
             earned_possible_values = []
-            for block_location in graded_scorable_blocks:
+            for block_location in context.graded_scorable_blocks_header:
```

Comment:
Does `context.graded_scorable_blocks_header` actually contains a block location? The name isn't suggesting that.

Reply:
its a dict where locations is the key and value is the header name. ie ( `{"bock-id": "Question 1 (earned)"}`